### PR TITLE
Fix bugs

### LINKED ISSUE #203: [PE-D] Name and category fields can contain only symbols
Name and category can contain ONLY symbols as shown by the third product

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: louisdavinlie/ped#5

### LINKED ISSUE #231: Price shows incorrect value when the number is really huge
**Describe the bug**

Price shows incorrect value when the number is really huge

**To Reproduce**

1. Create a product with a price of 8712635816758012375190823857109272034719278335

**Expected behavior**

The correct price should be shown

### LINKED ISSUE #234: Inconsistent width in popup
**Describe the bug**

The popup sometimes will have inconsistent width.

**To Reproduce**

1. Create two product, one with extremely long name, one with short name
2. Click the edit button for the first product with a long name
3. Click the edit button for the second product with short name

**Expected behavior**

The width for popup should be consistent

### LINKED ISSUE #235: Update Product Description Field with new line
**Describe the bug**

Error is not shown when the description does contains a new line.

**To Reproduce**

1. Create a new product with the popup
2. Add new lines in the description field

**Expected behavior**

An error should be shown

### PR DESCRIPTION
**Pull request by MechFroG88**
# Changes

1. Allow leading zero in all number field
2. Fix a limit on the price
3. Code refactoring

# Related Issues/ PRs
- Fixes #231 
- Fixes #235 
- Fixes #234 
- Fixes #203

### CODE REVIEW

File: src/main/java/seedu/ibook/model/item/Quantity.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
     /*
      * Must be a non-negative integer at most Quantity.MAX_QUANTITY.
      */
-    public static final String VALIDATION_REGEX = "0*\\d{1,6}";
+    public static final String VALIDATION_REGEX = "0*\\d{0,5}";
```
Comment (1001mei) [inline src/main/java/seedu/ibook/model/item/Quantity.java:17]: I think this should remain {1, 6}

Comment (MechFroG88) [reply src/main/java/seedu/ibook/model/item/Quantity.java]: Yeap, I have fixed that

File: src/main/java/seedu/ibook/model/item/Quantity.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
     /*
      * Must be a non-negative integer at most Quantity.MAX_QUANTITY.
      */
-    public static final String VALIDATION_REGEX = "0*\\d{1,6}";
+    public static final String VALIDATION_REGEX = "0*(?:0|[1-9]\\d{0,5})";
```
Comment (DavidTan0527) [inline src/main/java/seedu/ibook/model/item/Quantity.java:17]:
```suggestion
    public static final String VALIDATION_REGEX = "\\d{1,6}";
```

File: src/main/java/seedu/ibook/model/product/DiscountRate.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
     public static final String MESSAGE_CONSTRAINTS =
             "Discount Rate should only be a non-negative integer, and at most 100";
 
-    public static final String VALIDATION_REGEX = "|(?:0|[1-9]\\d?|100)%?";
+    public static final String VALIDATION_REGEX = "|0*(?:0|[1-9]\\d?|100)%?";
```
Comment (DavidTan0527) [inline src/main/java/seedu/ibook/model/product/DiscountRate.java:17]:
```suggestion
    public static final String VALIDATION_REGEX = "|(?:\\d{1,2}|100)%?";
```

File: src/main/java/seedu/ibook/model/product/Price.java [hunk 4]
```diff
@@ -9,10 +9,11 @@
  */
 public class Price {
 
+    public static final Double MAX_PRICE = 999999.99;
     public static final String MESSAGE_CONSTRAINTS =
-            "Prices should only have at most two decimal places, and should not be negative";
+            "Prices should only have at most two decimal places, non-negative and at most " + MAX_PRICE;
 
-    public static final String VALIDATION_REGEX = "\\$?(?:0|[1-9]\\d*)(?:\\.\\d{1,2})?";
+    public static final String VALIDATION_REGEX = "\\$?0*(?:0|[1-9]\\d{0,5})(?:\\.\\d{1,2})?";
```
Comment (DavidTan0527) [inline src/main/java/seedu/ibook/model/product/Price.java:16]:
```suggestion
    public static final String VALIDATION_REGEX = "\\$?\\d{1,6}(?:\\.\\d{1,2})?";
```

**Review by DavidTan0527 — COMMENTED** [review]
Just some regex simplifications. Else, LGTM.

**Pull request merged**